rbac/management/principal/it_service.py

```diff
@@ -315,6 +317,24 @@ def get_service_accounts_group(self, group: Group, user: User, options: dict = {
         else:
             return service_accounts
 
+    def generate_service_accounts_report_in_group(self, group: Group, client_ids: set[UUID]) -> dict[str, bool]:
+        """Check if the given service accounts are in the specified group."""
+        # Fetch the service accounts from the group.
+        group_service_account_principals = (
+            group.principals.values_list("service_account_id", flat=True)
+            .filter(type=TYPE_SERVICE_ACCOUNT)
+            .filter(service_account_id__in=client_ids)
```

Comment:
I believe we can supply a list of string directly in this query instead of making them into UUIDs, so it can save us some time converting back and forth. You can confirm by if that still passes your unit test.

And turn the group_service_account_principals into a set would help improving the speed when checking if the id is in it.

Reply:
Yeah, I wanted to be like as most type precise as possible. But it is true that it leads to many conversions left and right. I've the code that attempts to build UUIDs from the given client IDs, though, to make sure that we receive UUIDs and not any other type of string.